src/messaging.cc

```diff
@@ -278,15 +316,23 @@ Object* MessageDecoder::decode() {
   return null;
 }
 
-Object* MessageDecoder::decode_string() {
+Object* MessageDecoder::decode_string(bool inlined) {
   int length = read_cardinal();
-  uint8* data = read_pointer();
-  String* result = _process->object_heap()->allocate_external_string(length, data, true);
+  String* result = null;
+  if (inlined) {
+    ASSERT(length <= String::max_internal_size());
+    Error* error = null;
+    result = _process->allocate_string(reinterpret_cast<char*>(&_buffer[_cursor]), length, &error);
```

Comment:
Comment that we can ignore the error because we are below the max internal string size, so we know it's an internal allcoation error.